### PR TITLE
OJ-2334: Update state machine to dispatch response received event

### PR DESCRIPTION
**Pull request by Connor-Johnson-DCO**
### What changed

- State machine sends response received event to event bus
- Event rule created for response received event 

### Why did it change

- To capture events for TXMA

### Issue tracking

- [OJ-2334](https://govukverify.atlassian.net/browse/OJ-2334)

### Environment variables or secrets

- [ ] No environment variables or secrets were added or changed

- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

[OJ-2334]: https://govukverify.atlassian.net/browse/OJ-2334?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: infrastructure/template.yaml [hunk 1]
```diff
@@ -428,6 +436,12 @@ Resources:
         BearerTokenName: !Ref BearerTokenName
         CommonStackName: !Ref CommonStackName
         SsmParametersFunction: !Ref SsmParametersFunction
+        CheckHmrcEventBus: !Ref CheckHmrcEventBus
+        CheckHmrcEventBusSource: !Sub review-hc.${Environment}.account.gov.uk
+        AuditEventNameResponseReceived: !Sub
+          - ${EventPrefix}_{Event}
```
Comment (jkunle) [inline infrastructure/template.yaml:442]: Nice this is also a good alternative

File: infrastructure/template.yaml [hunk 2]
```diff
@@ -617,6 +633,20 @@ Resources:
...
+    Type: AWS::Events::Rule
+    Properties:
+      EventBusName: !Ref CheckHmrcEventBus
+      Description: Capture the audit event RESPONSE_RECEIVED
+      State: DISABLED
+      EventPattern:
+        source:
+          - !Sub
+            - review-hc.${Env}.account.gov.uk
+            - Env: !Ref Environment
```
Comment (jkunle) [inline infrastructure/template.yaml:646]: Is this really necessary `review-hc.${Environment}.account.gov.uk` should work fine

File: infrastructure/template.yaml [hunk 3]
```diff
@@ -428,6 +436,12 @@ Resources:
         BearerTokenName: !Ref BearerTokenName
         CommonStackName: !Ref CommonStackName
         SsmParametersFunction: !Ref SsmParametersFunction
+        CheckHmrcEventBus: !Ref CheckHmrcEventBus
+        CheckHmrcEventBusSource: !Sub review-hc.${Environment}.account.gov.uk
+        AuditEventNameResponseReceived: !Sub
+          - ${EventPrefix}_${Event}
+          - EventPrefix: !Ref AuditEventNamePrefix
+            Event: !FindInMap [Audit, EventName, ResponseReceived]
```
Comment (jkunle) [inline infrastructure/template.yaml:444]: is this missing - , i mean `- Event`?

File: infrastructure/template.yaml (hunk 2 above)
Comment (Connor-Johnson-DCO) [reply infrastructure/template.yaml:646]: No this changed in https://github.com/govuk-one-login/ipv-cri-check-hmrc-api/pull/154. I've created cascading PRs

File: infrastructure/template.yaml (hunk 3 above)
Comment (Connor-Johnson-DCO) [reply infrastructure/template.yaml:444]: No, this is the correct format from the [docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-sub.html)

**Review by jkunle — APPROVED** [review]

**Pull request merged**